lib/nodekey/NodeKey.ts

```diff
@@ -0,0 +1,95 @@
+import CryptoJS from 'crypto-js';
+import { randomBytes } from 'crypto';
+import secp256k1 from 'secp256k1';
+import fs from 'fs';
+
+/**
+ * A class representing an ECDSA public/private key pair that identifies an xud node on the network
+ * and can sign messages to prove their veracity.
+ */
+class NodeKey {
+  private pubKeyStr: string;
+
+  constructor(private privKey: Buffer) {
+    this.privKey = privKey;
+    const pubKey = secp256k1.publicKeyCreate(privKey);
+    this.pubKeyStr = pubKey.toString('hex');
+  }
+
+  /**
+   * Generate a random NodeKey.
+   */
+  static generate = (): NodeKey => {
+    let privKey: Buffer;
+    do {
+      privKey = randomBytes(32);
+    } while (!secp256k1.privateKeyVerify(privKey));
+
+    return new NodeKey(privKey);
+  }
+
+  /**
+   * Load a NodeKey from a file.
+   * @param path The path to the file
+   * @param password An optional password to decrypt the file
+   * @returns A node key if a file containing a valid ECDSA private key exists at the given path, otherwise undefined
+   */
+  static fromFile = (path: string, password?: string): NodeKey | undefined => {
+    try {
+      let buf: Buffer;
+      if (password) {
+        const encryptedString: string = fs.readFileSync(path, 'utf8');
+        const decryptedString: string = CryptoJS.AES.decrypt(encryptedString, password).toString(CryptoJS.enc.Hex);
+        buf = Buffer.from(decryptedString, 'hex');
+      } else {
+        buf = fs.readFileSync(path);
+      }
+
+      if (secp256k1.privateKeyVerify(buf)) {
+        return new NodeKey(buf);
+      } else {
+        throw new Error(`${path} does does not contain a valid ECDSA private key`);
+      }
+    } catch (err) {
+      if (err.code === 'ENOENT') {
+        return undefined;
```

Comment:
If I simply returned nothing, I got a warning because "not all code paths return a value." But I just realized if I change the return type to `NodeKey | void` it allows me to drop this statement, so I'll do that.

Reply:
I think you should always return `NodeKey`, and throw exception on error. Otherwise, we'll have to check if the object was initialized or not.

Why did you wanted to avoid throwing an exception if the error is specifically `ENOENT`?